### PR TITLE
:shell: Reception shell

### PR DESCRIPTION
**Pull request by landryarki**
added a shell function that loop around user input but that nothing atm.

renamed some stuff to make it coherent.

this is a draft for you to tell me how you prefer it to be implemented with the rest of the program

### CODE REVIEW

File: src/shellParser.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+{
+    std::string type;
+    while (!is_eof() && isalpha(_input[_pos])) {
+        type += _input[_pos];
+        _pos++;
+    }
+    return {Token::TYPE, type};
+}
+
+std::string sizes[] = {"S", "M", "L", "XL", "XXL"};
```
Comment (Thomaltarix) [inline src/shellParser.cpp:49]: It should go in an hpp imo

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
some early changes

File: include/shellParser.hpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        NUMBER = 4,
+        SEMICOLON = 8,
+        END = 16,
+        INVALID = 0
+    };
+
+    Token() : _type(INVALID), _value("") {}
+    Token(Type type, std::string value) : _type(type), _value(std::move(value)) {}
+    Token(const Token &token) : _type(token._type), _value(token._value) {}
+    Token &operator=(const Token &token)
```
[the remark this answers is not in the document]
Comment (landryarki) [reply include/shellParser.hpp:31]: :sleeping:

File: include/shellParser.hpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        if (this != &token) {
+            _type = token._type;
+            _value = token._value;
+        }
+        return *this;
+    }
+
+    Type _type;
+    std::string _value;
```
[the remark this answers is not in the document]
Comment (landryarki) [reply include/shellParser.hpp:41]: no, for later use in the program you should  be able to use them  (to make a getter ? not for me :sunglasses: )

[the remark this answers is not in the document]
Comment (landryarki) [reply include/shellParser.hpp]: nvm they are now

File: include/shellParser.hpp [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            /**
+             * @brief Constructor for the InvalidToken class.
+             * @param message The error message.
+             */
+            explicit InvalidToken(const std::string &message) : std::runtime_error(message) {}
+        };
+
+    private:
+        std::string _input;
+        size_t _pos;
```
Comment (Thomaltarix) [inline include/shellParser.hpp:153]: You can add comments here to

File: include/shellParser.hpp [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        [[ nodiscard ]] bool is_space() const { return _input[_pos] == ' ' || _input[_pos] == '\n' || _input[_pos] == '\t'; }
+        [[ nodiscard ]] bool is_eof() const { return _pos >= _input.size(); }
+
+        void skip_whitespace();
+
+        Token parseSemicolon();
+        Token parseType();
+        Token parseSize();
+        Token parseNumber();
+        std::vector<Token (Parser::*)()> _parsers = {&Parser::parseSemicolon, &Parser::parseSize, &Parser::parseNumber, &Parser::parseType};
```
Comment (Thomaltarix) [inline include/shellParser.hpp:166]: You can add comments here to

**Review by Thomaltarix — COMMENTED** [review]
You missed some code documentation, but overall, it's fine

**Comment by landryarki** [comment]
i did not comment private methods since i think this is not needed @Thomaltarix

**Comment by Thomaltarix** [comment]
It's not useful for the the devs since they definitely can't use this function, but it can be useful for us to understand what it does

**Review by Thomaltarix — APPROVED** [review]

**Review by Peralban — APPROVED** [review]

**Pull request merged**